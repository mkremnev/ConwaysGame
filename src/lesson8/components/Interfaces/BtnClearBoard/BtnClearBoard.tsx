import React, { FC } from 'react';
import { BtnClearBoardType } from '../../../types/Interfaces/BtnClearBoardType';

export const BtnClearBoard: FC<BtnClearBoardType> = ({ onClick }) => {
	return (
		<button className="btn btn-clear" onClick={onClick}>
			{'Очистить'}
		</button>
	);
};
