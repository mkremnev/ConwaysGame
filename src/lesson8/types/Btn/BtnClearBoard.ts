export type BtnClearBoard = {
	onClick: () => false;
};
