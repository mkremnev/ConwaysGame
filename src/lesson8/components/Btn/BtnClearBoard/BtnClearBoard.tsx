import React, { FC } from 'react';
import { BtnClearBoard } from '../../../types/Btn/BtnClearBoard';

export const BtnClearBord: FC<BtnClearBoard> = ({ onClick }) => {
	return (
		<button className="btn btn-clear" onClick={() => onClick()}></button>
	);
};
