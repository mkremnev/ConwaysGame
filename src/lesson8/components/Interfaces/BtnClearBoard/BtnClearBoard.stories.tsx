import React from 'react';
import { BtnClearBoard } from './BtnClearBoard';
import { action } from '@storybook/addon-actions';
import { withKnobs } from '@storybook/addon-knobs';

// eslint-disable-next-line no-restricted-syntax
export default {
	title: 'Lesson 8 / Interfaces / BtnClearBoard',
	decorators: [withKnobs],
};

export const BtnClearBoardInterfaces = () => [
	<BtnClearBoard onClick={action('Button clicked')} key="interfaces" />,
];
