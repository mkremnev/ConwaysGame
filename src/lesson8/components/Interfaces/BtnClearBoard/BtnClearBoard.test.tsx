import React from 'react';
import { mount } from 'enzyme';
import { BtnClearBoard } from './BtnClearBoard';
import renderer from 'react-test-renderer';

describe('BtnClearBoard', () => {
	it('renders button clear board', () => {
		expect(
			renderer.create(<BtnClearBoard onClick={jest.fn()} />).toJSON(),
		).toMatchSnapshot();
	});
	it('calls onClick callback on click btn', () => {
		const onClick = jest.fn();
		const wrapper = mount(<BtnClearBoard onClick={onClick} />);
		wrapper.simulate('click');
		expect(onClick).toHaveBeenCalled();
	});
});
