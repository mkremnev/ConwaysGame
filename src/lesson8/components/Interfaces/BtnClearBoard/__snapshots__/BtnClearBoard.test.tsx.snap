// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnClearBoard renders button clear board 1`] = `
<button
  className="btn btn-clear"
  onClick={[MockFunction]}
>
  Очистить
</button>
`;
